src/container/srv_target.c

```diff
@@ -1667,6 +1676,32 @@ ds_cont_tgt_close(uuid_t hdl_uuid)
 	return dss_thread_collective(cont_close_one_hdl, &arg, 0);
 }
 
+static int
+cont_force_close_cb(d_list_t *rlink, void *arg)
+{
+	struct coll_close_arg	*cca = arg;
+	struct ds_cont_hdl	*hdl = cont_hdl_obj(rlink);
+
+	if (hdl->sch_cont == NULL || uuid_compare(hdl->sch_cont->sc_pool_uuid, cca->uuid) != 0)
+		return 0;
+
+	D_DEBUG(DF_DSMS, "Force closing "DF_UUID"/"DF_UUID"/"DF_UUID"\n",
+		DP_UUID(cca->uuid), DP_UUID(hdl->sch_cont->sc_uuid), DP_UUID(hdl->sch_uuid));
+
+	return cont_close_hdl(hdl->sch_uuid);
+}
+
+/* Force close all containers for the given pool. */
+static int
+ds_cont_child_close_all(uuid_t po_uuid)
+{
+	struct dsm_tls		*tls = dsm_tls_get();
+	struct coll_close_arg	 cca;
+
+	uuid_copy(cca.uuid, po_uuid);
+	return d_hash_table_traverse(&tls->dt_cont_hdl_hash, cont_force_close_cb, &cca);
```

Comment:
d_hash_table_traverse() doesn't support removal.

Reply:
Since then, we may have to consider to establish the bi-relationship between cont_child and the ds_cont_hdl.